presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java

```diff
@@ -272,4 +272,17 @@ public void testFilteringForTablesAndColumns()
         actualColumns = computeActual("DESCRIBE " + datasourceB);
         Assert.assertEquals(actualColumns, expectedColumns);
     }
+
+    @Test
+    public void testLimitPushdown()
+    {
+        assertThat(query("SELECT name FROM nation LIMIT 30")).isCorrectlyPushedDown(); // Use high limit for result determinism
+
+        // with filter over numeric column
+        assertThat(query("SELECT name FROM nation WHERE regionkey = 3 LIMIT 5")).isCorrectlyPushedDown();
+
+        // with filter over varchar column
+        assertThat(query("SELECT name FROM nation WHERE name < 'EEE' LIMIT 5")).isCorrectlyPushedDown();
+    }
```

Comment:
what about aggregation and project? We have at the moment 4 variables (predicate, limit, projection and aggregation). 
Having dedicated tests per each pushdown gives 4 tests, for pairs another 12 tests, for triples another test 4 tests, and for all cases at once another 1 tests. 21 in total.

What are your thoughts to tackle that?

Reply:
> what about aggregation and project?

agg pushdown with limit does not work today -- https://github.com/prestosql/presto/issues/5261

> What are your thoughts to tackle that?

i want to organize pushdown test around top level thing that is pushed down (eg agg pushdown tests won't cover LIMIT (applied after agg), unless the LIMIT is in subquery)
i also want to order them by the ordering in which they are appledi in JdbcTableHandle